File: backend/store-management/src/main/java/com/storemanagement/service/ShippingAddressService.java
package com.storemanagement.service;

import com.storemanagement.dto.shipment.ShippingAddressDTO;
import com.storemanagement.dto.shipment.CreateShippingAddressRequestDto;
import com.storemanagement.dto.shipment.UpdateShippingAddressRequestDto;

import java.util.List;

public interface ShippingAddressService {
    List<ShippingAddressDTO> getAllAddresses(Integer customerId);
    
    ShippingAddressDTO getDefaultAddress(Integer customerId);
    
    ShippingAddressDTO createAddress(Integer customerId, CreateShippingAddressRequestDto request);
    
    ShippingAddressDTO updateAddress(Integer customerId, Integer addressId, UpdateShippingAddressRequestDto request);
    
    ShippingAddressDTO setDefaultAddress(Integer customerId, Integer addressId);
    
    void deleteAddress(Integer customerId, Integer addressId);
}

File: frontend_client/src/services/shippingAddressService.js
import api from './api';
import { API_ENDPOINTS } from '../constants/apiEndpoints';

const unwrap = (resp) => resp?.data ?? resp;

export const shippingAddressService = {

  getAllAddresses: async () => {
    const resp = await api.get(API_ENDPOINTS.SHIPPING_ADDRESSES.BASE);
    return unwrap(resp);
  },

  getDefaultAddress: async () => {
    const resp = await api.get(API_ENDPOINTS.SHIPPING_ADDRESSES.DEFAULT);
    return unwrap(resp);
  },

  createAddress: async (addressData) => {
    const resp = await api.post(API_ENDPOINTS.SHIPPING_ADDRESSES.BASE, addressData);
    return unwrap(resp);
  },

  updateAddress: async (addressId, addressData) => {
    const resp = await api.put(API_ENDPOINTS.SHIPPING_ADDRESSES.BY_ID(addressId), addressData);
    return unwrap(resp);
  },

  setDefaultAddress: async (addressId) => {
    const resp = await api.put(API_ENDPOINTS.SHIPPING_ADDRESSES.SET_DEFAULT(addressId));
    return unwrap(resp);
  },

  deleteAddress: async (addressId) => {
    const resp = await api.delete(API_ENDPOINTS.SHIPPING_ADDRESSES.BY_ID(addressId));
    return unwrap(resp);
  },
};
